SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+
+## Backwards Compatibility
+<!--All SWIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The SWIP must explain how the author proposes to deal with these incompatibilities. SWIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+Currently, Swarm is not running with a live testnet for setting prices. Therefore, no compatibility problems are expected if the SWIP is implemented before the currently implemented Swarm incentivisation goes live. Should that not be the case and this SWIP is adopted at a later stage, non-up to date nodes may be over time dropped due to different honey prices.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for SWIPs that are affecting changes to data and message formats. Other SWIPs can choose to include links to test cases if applicable.-->
+Not currently available
+ 
+## Implementations
+<!--The implementations must be completed before any SWIP is given status "Final", but it need not be completed before the SWIP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.-->
+* Swarm's initial implementation of the `honeyTo MoneyOracle` is composed of two pieces: a `honeyToMoneyContract` (which runs as a smart-contract on the Ethereum blockchain), and a `honeyToMoneyOracleWrapper` (which runs locally by the node). The responsibility of the `honeyToMoneyContract` is to emit all relevant information, while the responsibility of the `honeyToMoneyOracleWrapper` is to piece all this information together into a structure which is useful for the node. 
+* The Swarm `honeyToMoneyOracle` will reference an Ethereum address of a `HoneyToMoney` price oracle smart contract. This smart contract will fulfill the role of the external oracle.
+* The currency in which the oracle quotes its price is Wei.
+* The `honeyToMoneyContract` implements the `HoneyToMoney` interface (to be specified). 
+* The `HoneyToMoneyContract` will initially be governed in such a way that a subset of stakeholders can make decisions which are not high-impact (i.e. decide on small price changes) but the involvement of all stakeholders is required to make high-impact decisions (such as changing the implementation of the oracle, changing governance, deciding on big price changes, etc). This semi-governance structure is in place to allow experimentation with an update process of honey prices and is not a recommendation for a price update process when honey is quoted in a currency worth real money.
```

Comment:
I am sure people would be very interested in how this governance scheme will take place. Is this a separate SWIP? Since you mention other what elements are separate SWIPs other places in this document, plesae consider whether it's worth a mention here, too.

Reply:
@nolash , I started working on an implementation of the simple governance scheme described here [see here](https://github.com/Eknir/msgOracle). AFAIK, there is no intention to write a SWIP for this yet, as the idea is to first get some experience. 

When we are going to run Swarm on a live network, there must be a clear idea how the price is going to be updated, what the process around this is (at that time) and how we envision updating this process to give more autonomy to the users of Swarm.